inference-engine/thirdparty/clDNN/src/memory_pool.cpp

```diff
@@ -141,6 +141,77 @@ bool memory_pool::has_conflict(const memory_set& a,
     return !intersection.empty();
 }
 
+void memory_pool::release_memory(memory_impl* mem,
+    const primitive_id& id) {
+    // check nonpadded pool first
+    auto _layout = mem->get_layout();
+    auto type = mem->get_allocation_type();
+    auto network_id = mem->get_net_id();
+
+    {
+        auto range = _non_padded_pool.equal_range(_layout.bytes_count());
+        auto it = range.first;
+
+        while (it != range.second && it != _non_padded_pool.end()) {
```

Comment:
It seems that the second part of the condition is not needed, because upper_bound (`range.second`) of map's end will be the same as `_non_padded_pool.end()`

Reply:
Since range is obtained before erase() call, its iterators might become invalidated. So I decided to be safe than sorry here.